### DISCUSSION TITLE
When do module scoped fixtures run?

### DISCUSSION THREAD

**Discussion by jasonhoppessf** [pytest-dev/pytest#9998] [General]
I am new to pytest and I have seen two contradictory examples online. Do module scoped fixtures run as soon as the module is loaded or when passed to a method? If as soon as the module is loaded how do I determine the order? One example is here with grpc_add_to_server  [https://pypi.org/project/pytest-grpc/](url). The other example is from the documentation [https://docs.pytest.org/en/6.2.x/fixture.html#higher-scoped-fixtures-are-executed-first](url).

**Comment by nicoddemus** [comment]
Hi @jasonhoppessf,

They execute as soon as the first test in a module **requests** it, and are deleted later when the last test of the module finishes (regardless if it requested the module-scoped fixture or not) [^1]. 


[^1]: Well not technically 100% this part, but it is simpler to think it like this.
